File: sdk/python/pageiq_client.py
# PageIQ Python SDK

import requests
from typing import Dict, List, Optional, Union

class PageIQClient:
    """Python client for PageIQ API"""

    def __init__(self, api_key: str, base_url: str = "https://pageiq.p.rapidapi.com"):
        """
        Initialize the PageIQ client

        Args:
            api_key: Your RapidAPI key
            base_url: API base URL (default: RapidAPI endpoint)
        """
        self.api_key = api_key
        self.base_url = base_url.rstrip('/')
        self.session = requests.Session()
        self.session.headers.update({
            'X-RapidAPI-Key': api_key,
            'X-RapidAPI-Host': 'pageiq.p.rapidapi.com',
            'Content-Type': 'application/json'
        })

    def analyze_website(
        self,
        url: str,
        screenshot: bool = False,
        use_browser: bool = False
    ) -> Dict:
        """
        Analyze a single website

        Args:
            url: Website URL to analyze
            screenshot: Include screenshot (Pro plan required)
            use_browser: Use browser automation for JS-heavy sites

        Returns:
            Analysis result dictionary
        """
        payload = {
            "url": url,
            "options": {
                "screenshot": screenshot,
                "use_browser": use_browser
            }
        }

        response = self.session.post(f"{self.base_url}/api/v1/analyze", json=payload)
        response.raise_for_status()
        return response.json()

    def batch_analyze_websites(
        self,
        urls: List[str],
        webhook_url: Optional[str] = None,
        screenshot: bool = False
    ) -> Dict:
        """
        Analyze multiple websites in batch

        Args:
            urls: List of URLs to analyze
            webhook_url: Optional webhook URL for completion notification
            screenshot: Include screenshots

        Returns:
            Batch analysis response
        """
        payload = {
            "urls": urls,
            "options": {"screenshot": screenshot}
        }

        if webhook_url:
            payload["webhook_url"] = webhook_url

        response = self.session.post(f"{self.base_url}/api/v1/batch-analyze", json=payload)
        response.raise_for_status()
        return response.json()

    def get_batch_status(self, batch_id: str) -> Dict:
        """
        Get status of batch analysis

        Args:
            batch_id: Batch ID from batch_analyze_websites

        Returns:
            Batch status information
        """
        response = self.session.get(f"{self.base_url}/api/v1/batch-analyze/{batch_id}")
        response.raise_for_status()
        return response.json()

    def get_usage_stats(self) -> Dict:
        """
        Get API usage statistics

        Returns:
            Usage statistics
        """
        response = self.session.get(f"{self.base_url}/api/v1/usage")
        response.raise_for_status()
        return response.json()


# Example usage
if __name__ == "__main__":
    client = PageIQClient("your-rapidapi-key-here")

    # Analyze single website
    try:
        result = client.analyze_website("https://example.com", screenshot=True)
        print("Analysis result:", result)
    except Exception as e:
        print("Error:", e)

    # Batch analysis
    try:
        batch_result = client.batch_analyze_websites([
            "https://example.com",
            "https://github.com"
        ], webhook_url="https://your-webhook-url.com/notify")
        print("Batch started:", batch_result)

        # Check status later
        import time
        time.sleep(30)  # Wait for processing
        status = client.get_batch_status(batch_result["data"]["batch_id"])
        print("Batch status:", status)
    except Exception as e:
        print("Error:", e)

File: sdk/javascript/pageiq-client.js
/**
 * PageIQ JavaScript SDK
 * For use with RapidAPI
 */

class PageIQClient {
    constructor(apiKey, baseURL = 'https://pageiq.p.rapidapi.com') {
        this.apiKey = apiKey;
        this.baseURL = baseURL.replace(/\/$/, '');

        this.defaultHeaders = {
            'X-RapidAPI-Key': apiKey,
            'X-RapidAPI-Host': 'pageiq.p.rapidapi.com',
            'Content-Type': 'application/json'
        };
    }

    /**
     * Analyze a single website
     * @param {string} url - Website URL to analyze
     * @param {Object} options - Analysis options
     * @param {boolean} options.screenshot - Include screenshot (Pro plan)
     * @param {boolean} options.useBrowser - Use browser automation
     * @returns {Promise<Object>} Analysis result
     */
    async analyzeWebsite(url, options = {}) {
        const payload = {
            url: url,
            options: {
                screenshot: options.screenshot || false,
                use_browser: options.useBrowser || false,
                ...options
            }
        };

        const response = await fetch(`${this.baseURL}/api/v1/analyze`, {
            method: 'POST',
            headers: this.defaultHeaders,
            body: JSON.stringify(payload)
        });

        if (!response.ok) {
            throw new Error(`API request failed: ${response.status} ${response.statusText}`);
        }

        return await response.json();
    }

    /**
     * Analyze multiple websites in batch
     * @param {string[]} urls - Array of URLs to analyze
     * @param {Object} options - Batch options
     * @param {string} options.webhookUrl - Webhook URL for notifications
     * @param {boolean} options.screenshot - Include screenshots
     * @returns {Promise<Object>} Batch analysis response
     */
    async batchAnalyzeWebsites(urls, options = {}) {
        const payload = {
            urls: urls,
            options: {
                screenshot: options.screenshot || false,
                ...options
            }
        };

        if (options.webhookUrl) {
            payload.webhook_url = options.webhookUrl;
        }

        const response = await fetch(`${this.baseURL}/api/v1/batch-analyze`, {
            method: 'POST',
            headers: this.defaultHeaders,
            body: JSON.stringify(payload)
        });

        if (!response.ok) {
            throw new Error(`API request failed: ${response.status} ${response.statusText}`);
        }

        return await response.json();
    }

    /**
     * Get batch analysis status
     * @param {string} batchId - Batch ID from batch analysis
     * @returns {Promise<Object>} Batch status
     */
    async getBatchStatus(batchId) {
        const response = await fetch(`${this.baseURL}/api/v1/batch-analyze/${batchId}`, {
            method: 'GET',
            headers: this.defaultHeaders
        });

        if (!response.ok) {
            throw new Error(`API request failed: ${response.status} ${response.statusText}`);
        }

        return await response.json();
    }

    /**
     * Get health status
     * @returns {Promise<Object>} Health status
     */
    async getHealthStatus() {
        const response = await fetch(`${this.baseURL}/health`, {
            method: 'GET',
            headers: this.defaultHeaders
        });

        if (!response.ok) {
            throw new Error(`API request failed: ${response.status} ${response.statusText}`);
        }

        return await response.json();
    }
}

// Example usage with promises
function exampleUsage() {
    const client = new PageIQClient('your-rapidapi-key-here');

    // Analyze single website
    client.analyzeWebsite('https://example.com', { screenshot: true })
        .then(result => {
            console.log('Analysis result:', result);
        })
        .catch(error => {
            console.error('Error:', error);
        });

    // Batch analysis with webhook
    client.batchAnalyzeWebsites([
        'https://example.com',
        'https://github.com',
        'https://stackoverflow.com'
    ], {
        webhookUrl: 'https://your-webhook-url.com/notify',
        screenshot: false
    })
    .then(batchResult => {
        console.log('Batch started:', batchResult);

        // Check status after some time
        setTimeout(() => {
            client.getBatchStatus(batchResult.data.batch_id)
                .then(status => console.log('Batch status:', status))
                .catch(error => console.error('Status check error:', error));
        }, 30000);
    })
    .catch(error => {
        console.error('Batch error:', error);
    });
}

// Example usage with async/await
async function asyncExample() {
    const client = new PageIQClient('your-rapidapi-key-here');

    try {
        const result = await client.analyzeWebsite('https://example.com');
        console.log('Result:', result);
    } catch (error) {
        console.error('Error:', error);
    }
}

// Export for Node.js or browser
if (typeof module !== 'undefined' && module.exports) {
    module.exports = PageIQClient;
} else if (typeof window !== 'undefined') {
    window.PageIQClient = PageIQClient;
}